dotnet/src/Connectors/Connectors.OpenAI/Core/OpenAIImageContent.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+using OpenAI.Chat;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// OpenAI specialized image content.
+/// </summary>
+[Experimental("SKEXP0010")]
+public sealed class OpenAIImageContent : ImageContent
+{
+    /// <summary>
+    /// Gets the metadata key for the detail level property.
+    /// </summary>
+    public const string DetailLevelProperty = nameof(DetailLevel);
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenAIImageContent"/> class.
+    /// </summary>
+    [JsonConstructor]
+    public OpenAIImageContent()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenAIImageContent"/> class.
+    /// </summary>
+    /// <param name="uri">The URI of image.</param>
+    public OpenAIImageContent(Uri uri) : base(uri)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenAIImageContent"/> class.
+    /// </summary>
+    /// <param name="dataUri">DataUri of the image</param>
+    public OpenAIImageContent(string dataUri) : base(dataUri)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OpenAIImageContent"/> class.
+    /// </summary>
+    /// <param name="data">Byte array of the image</param>
+    /// <param name="mimeType">Mime type of the image</param>
+    public OpenAIImageContent(ReadOnlyMemory<byte> data, string? mimeType) : base(data, mimeType)
+    {
+    }
+
+    /// <summary>
+    /// The level of detail with which the model should process the image and generate its textual understanding of it.
+    /// More information here: <see href="https://platform.openai.com/docs/guides/vision/low-or-high-fidelity-image-understanding" />.
+    /// </summary>
+    public ChatImageDetailLevel? DetailLevel { get; set; }
```

Comment:
1. We need to check if ChatImageDetailLevel is serializable for scenarios involving chat history serialization/deserialization.   
2. Most likely, the value of the property will be lost during the chat history serialization/deserialization round-trip because the `OpenAIImageContent` will be serialized as ImageContent. A similar issue with OpenAI functions was solved [by placing them in the Metadata dictionary](https://github.com/microsoft/semantic-kernel/blob/3332079da920bc5da608bb97a441efeffc7c496b/dotnet/src/Connectors/Connectors.OpenAI/Core/OpenAIChatMessageContent.cs#L127), which is part of the base KernelContent class and is always serialized/deserialized.

Reply:
Good catch! It appeared that `ChatImageDetailLevel` is not serializable, so I removed the usage of it to avoid problems with {de}serialization and keep the usage of `ImageContent.Metadata` only.